SpatialGDK/Source/SpatialGDKFunctionalTests/Public/SpatialFunctionalTest.h

```diff
@@ -403,6 +409,53 @@ class SPATIALGDKFUNCTIONALTESTS_API ASpatialFunctionalTest : public AFunctionalT
 	// Clears all the snapshots taken, not meant to be used directly.
 	static void ClearAllTakenSnapshots();
 
+	// Members used for tests that are also their own standalone map.
+
+	virtual void CreateCustomContentForMap() {}
+
+	void GenerateMap() override;
+
+	bool ShouldGenerateMap() override;
+
+	bool SaveMap() override;
+
+	template <class T>
+	T& AddActor(const FTransform& Transform = FTransform::Identity)
+	{
+		checkf(bIsGeneratingMap, TEXT("AddActor should only be called from within an overridden CreateCustomContentForMap."));
+		return GeneratedTestMap->AddActorToLevel<T>(GeneratedTestMap->GetWorld()->GetCurrentLevel(), Transform);
+	}
+
+	bool GenerateCustomConfig() override;
+
+	FString GetMapName() override;
+
+protected:
+	// Derived tests can call this to set the string that will be printed into the .ini file to be used with this map to override
+	// settings specifically for this test map. Should be called during constructor.
+	void SetCustomConfig(FString& String);
+
+	ASpatialWorldSettings* GetWorldSettings();
+
+	// The constructor is the intended way to set the following settings, however these setters can be used if a test is not a direct
+	// descendant of ASpatialFunctionalTest and needs to modify these settings.
+	void SetMapCiCategory(const EMapCategory MapCiCategory);
```

Comment:
I am not sure I understand the need for these functions? I don't think I get the comment, what use case are you making these for?

Reply:
Removed these in favour of passing the arguments up the constructor tree